src/settings/toml/manifest.rs

```diff
@@ -312,11 +312,32 @@ impl Manifest {
         environment_name: Option<&str>,
         preview: bool,
     ) -> Result<Target, failure::Error> {
-        // Site projects are always webpack for now; don't let toml override this.
-        let target_type = match self.site {
-            Some(_) => TargetType::Webpack,
-            None => self.target_type.clone(),
-        };
+        if self.site.is_some() {
+            match self.target_type {
+                TargetType::Rust => {
+                    failure::bail!(format!(
+                        "{} Workers Sites does not support Rust type projects.",
+                        emoji::WARN
+                    ))
+                }
+                TargetType::JavaScript => {
+                    if let Some(build) = &self.build {
+                        if build.command.is_none() {
+                            failure::bail!(format!(
```

Comment:
nit: would be nice if this message wasn't duplicated but I'm not too bothered by it if it's inconvenient to structure the code that way